.travis.yml

```diff
@@ -17,30 +17,59 @@ env:
     #   See https://docs.travis-ci.com/api#external-apis.
     - secure: "qQ3vMSXUybUyKljJjgcVT9YaWcZ0PkdLZSFj6BVuo82SWO7e4L+hC8pqsa1Sm/EjyetVwfKk5lkvruPpro++8hDPLq3ahNOMmm4axfcWvACpW+acD/VCA5D8Dxz2ohrvpwsqo7BmIV6OUtIO5JeePHP1SqaIdVZFJ0pIAZToEWQ="
     - CONDA_ROOT_PYTHON_VERSION: "2.7"
-    - NUMPY_VERSION=1.11.1
-    - PANDAS_VERSION=0.18.1
-    - SCIPY_VERSION=0.17.1
+  matrix:
+    - PANDAS_18=1
+    - PANDAS_22=1
 matrix:
+  exclude:
+    - python: 2.7
+      env: PANDAS_22=1
   include:
-    # - python: 3.5
-    #   env: PANDAS_VERSION=0.19.2
-    - python: 3.5
-      env: NUMPY_VERSION=1.14.1 PANDAS_VERSION=0.22.0 SCIPY_VERSION=1.0.0 PANDAS_DATAREADER_VERSION=0.4.0 DASK_VERSION=0.17.1
+    # Workaround Travis OSX not natively supporting Python.
+    - os: osx
+      language: generic
+      env: TRAVIS_PYTHON_VERSION=2.7 PANDAS_18=1
+    - os: osx
+      language: generic
+      env: TRAVIS_PYTHON_VERSION=3.5 PANDAS_18=1
+    - os: osx
+      language: generic
+      env: TRAVIS_PYTHON_VERSION=3.5 PANDAS_22=1
+
 cache:
   directories:
     - $HOME/.cache/.pip/
 
 before_install:
-  - if [ ${CONDA_ROOT_PYTHON_VERSION:0:1} == "2" ]; then wget https://repo.continuum.io/miniconda/Miniconda2-4.3.30-Linux-x86_64.sh -O miniconda.sh; else wget https://repo.continuum.io/miniconda/Miniconda3-4.3.30-Linux-x86_64.sh -O miniconda.sh; fi
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then MINICONDA_OS=MacOSX; else MINICONDA_OS=Linux; fi
```

Comment:
Thoughts on moving this into a script like `etc/travis_before_install.sh`? Writing bash in a yaml list seems strictly worse to me than writing it in a regular file that we invoke from the yaml.

Reply:
+1 on this

Reply:
What do you think about breaking out the different pieces of this section into `ci/travis/install_miniconda.sh` and `ci/travis/overwrite_requirements.sh` as opposed to one bigger script?

Reply:
That seems fine to me as well. Having a few lines in the yaml doesn't bother me. It just seems a bit silly to have a full blown bash script here.